src/amd/compiler/aco_live_var_analysis.cpp

```diff
@@ -244,8 +244,7 @@ live live_var_analysis(Program* program,
       unsigned block_idx = *b_it;
       worklist.erase(block_idx);
       process_live_temps_per_block(program, result, &program->blocks[block_idx], worklist);
-      new_demand.vgpr = std::max(new_demand.vgpr, program->blocks[block_idx].vgpr_demand);
-      new_demand.sgpr = std::max(new_demand.sgpr, program->blocks[block_idx].sgpr_demand);
+      new_demand.update_pressure(RegisterDemand(program->blocks[block_idx].vgpr_demand, program->blocks[block_idx].sgpr_demand));
```

Comment:
RegisterDemand() is not explicit, so I'd say {vgpr_demand, sgpr_demand} is sufficient here.

Reply:
It will be cleanup up in a follow-up commit when `aco::Block` gets a RegisterDemand rather than the two `int16_t`